packages/jupyter-ai/jupyter_ai/extension.py

```diff
@@ -460,37 +454,34 @@ def _init_chat_handlers(self, ychat: YChat) -> Dict[str, BaseChatHandler]:
         assert self.serverapp
 
         eps = entry_points()
-        chat_handler_eps = eps.select(group="jupyter_ai.chat_handlers")
+        all_chat_handler_eps = eps.select(group="jupyter_ai.chat_handlers")
+
+        # Override native chat handlers if duplicates are present
+        sorted_eps = sorted(
+            all_chat_handler_eps, key=lambda ep: ep.dist.name != "jupyter_ai"
+        )
+        seen = {}
+        for ep in sorted_eps:
+            seen[ep.name] = ep
+        chat_handler_eps = list(seen.values())
+
```

Comment:
I think it is important to avoid very long method definitions for readability. Can this block to moved to a new function inside new module, e.g. `get_chat_handler_eps()` under `jupyter_ai/entry_points/utils.py`?

Reply:
Can be done later.